src/ir_def/ir.ml

```diff
@@ -72,7 +72,7 @@ and exp' =
   | FuncE of                                   (* function *)
       string * Type.func_sort * Type.control * typ_bind list * arg list * Type.typ list * exp
   | SelfCallE of Type.typ list * exp * exp * exp (* essentially ICCallPrim (FuncE shared…) *)
-  | ActorE of dec list * field list * system * Type.typ (* actor *)
+  | ActorE of dec list * field list * system * Type.typ * exp (* actor *)
```

Comment:
Hmm, I wonder if a slightly cleaner refactoring might have been to put the additional type and exp fields in `ActorE` and `ActorU` into the `system` record. Don't bother now, it's just a note to self.

Reply:
I agree. I added a comment and made a temporary small code refactoring making it more obvious what `exp` stands for. We can then later combine it with `system`.